zmon_agent/discovery/kubernetes/cluster.py

```diff
@@ -1129,3 +1142,67 @@ def get_postgresql_databases(cluster_id, alias, environment, region, infrastruct
                 entities.append(replicatype_entity)
 
     return entities
+
+
+@trace(tags={'kubernetes': 'hpa'}, pass_span=True)
+def get_cluster_hpas(kube_client, cluster_id, alias, environment, region, infrastructure_account,
+                     namespace=None, **kwargs) -> list:
+    current_span = extract_span_from_kwargs(**kwargs)
+    entities = []
+
+    hpas = get_all(kube_client, kube_client.get_hpas, namespace, span=current_span)
+
+    for h in hpas:
+        obj = h.obj
+
+        entity = {
+            'id': 'hpa-{}-{}[{}]'.format(h.name, h.namespace, cluster_id),
+            'type': HPA_TYPE,
+            'kube_cluster': cluster_id,
+            'alias': alias,
+            'environment': environment,
+            'created_by': AGENT_TYPE,
+            'infrastructure_account': infrastructure_account,
+            'region': region,
+
+            'hpa_name': h.name,
+            'hpa_namespace': obj['metadata']['namespace'],
+
+            'hpa_desired_replicas': obj['status']['desiredReplicas'],
+            'hpa_current_replicas': obj['status']['currentReplicas']
+        }
+
+        entity.update(entity_labels(obj, 'labels', 'annotations'))
```

Comment:
I suggest we parse the `conditions` from the current stupid representation and expose them in better way (e.g. either unrolling them into individual fields such as `able_to_scale: true|false` or at least representing them properly).